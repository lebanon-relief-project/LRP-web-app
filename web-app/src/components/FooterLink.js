import React from "react";
import styled from "styled-components";
import devices from "../styles/Devices";

const FooterLink = (props) => {
  const links = props.links.map((link, index) => (
    <StyledListLink key={index}>{link}</StyledListLink>
  ));
  return (
    <StyledFooterLink>
      <FooterLinkTitle>{props.title}</FooterLinkTitle>
      <StyledList>{links}</StyledList>
    </StyledFooterLink>
  );
};

const FooterLinkTitle = styled.h2`
  font-weight: bold;
`;

const StyledFooterLink = styled.div`
  padding: 1rem;
  margin: 2rem 0;
  @media (max-width: ${devices.mobile}) {
    width: 75%;
    padding: 0;
    margin: 0;
    order: 2;
  }
`;

const StyledList = styled.ul`
  list-style-type: none;
  padding: 0;
`;

const StyledListLink = styled.li`
  text-decoration-line: underline;
`;

export default FooterLink;
