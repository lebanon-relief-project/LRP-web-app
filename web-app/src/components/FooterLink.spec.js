import React from "react";
import { render } from "@testing-library/react";
import FooterLink from "./FooterLink";
import "jest-styled-components";

describe("the FooterLink component", () => {
  let container;
  it("should match the snapshot", () => {
    ({ container } = renderFooterLink());

    expect(container).toMatchSnapshot();
  });
});

const renderFooterLink = () => {
  return render(<FooterLink title="testTitle" links={["link1", "link2"]} />);
};
