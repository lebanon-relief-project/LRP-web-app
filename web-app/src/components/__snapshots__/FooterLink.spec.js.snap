// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the FooterLink component should match the snapshot 1`] = `
.c1 {
  font-weight: bold;
}

.c0 {
  padding: 1rem;
  margin: 2rem 0;
}

.c2 {
  list-style-type: none;
  padding: 0;
}

.c3 {
  -webkit-text-decoration-line: underline;
  text-decoration-line: underline;
}

@media (max-width:576px) {
  .c0 {
    width: 75%;
    padding: 0;
    margin: 0;
    -webkit-order: 2;
    -ms-flex-order: 2;
    order: 2;
  }
}

<div>
  <div
    class="c0"
  >
    <h2
      class="c1"
    >
      testTitle
    </h2>
    <ul
      class="c2"
    >
      <li
        class="c3"
      >
        <a
          rel="noopener noreferrer"
          style="text-decoration: none;"
          target="_blank"
        />
      </li>
      <li
        class="c3"
      >
        <a
          rel="noopener noreferrer"
          style="text-decoration: none;"
          target="_blank"
        />
      </li>
    </ul>
  </div>
</div>
`;
